handler.go

```diff
@@ -40,6 +40,12 @@ func (h *handler) Handle(reqURL []string, r *http.Request) *response.Response {
 		return response.New(err.Error()).
 			Status(http.StatusBadRequest)
 	}
+
+	if err := h.verify(ctx, r); err != nil {
+		return response.New(err.Error()).
+			Status(http.StatusBadRequest)
```

Comment:
I prefer using wrap structure more, but I think we should do some PROTOTYPE to sure it.
Maybe more validator function is not a bad thing, the only thing I affair is users might misuse them since we just let them use context itself. I think VerifyError structure might also use response? Just implement `error` interface, and panic if the flag is not set?

Reply:
I read some https://rocket.rs/v0.4/guide/requests/#request-guards, and I found an __unused__ context really makes me confused, because we have a context, but never access it, it would make code be hard to read for newer of the framework.

Here has two ideas,
- inject a function `func validator(r *http.Request) error` on the handler, the benefit is easy, also easy for reuse on the different handler, the problem of it is we let user using `http` directly, so they could use the validator as __Fairing__.
- inject context into handler, but not on handling function, but a builder pattern, by this way, users can't touch `*http.Request` structure, and also could reuse the validator, but the concept would be more complex, and handler setup process would be more complex.

Reply:
It might look like:
```go
rocket.Get("/", func() string { return "hello" }).
    Guards(&User{}, &Cookie{}, ...) // or more
```